.github/workflows/maven.yml

```diff
@@ -114,12 +114,51 @@ jobs:
           name: jdk-${{ matrix.java }}-reports
           path: target/it-tests/*/build.log
 
+  test-mvn:
+    needs: linux
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false # Don't fail unfinished builds until we know which versions are compatible
+      matrix:
+        mvn: ['3.8.5', '3.6.3', '3.5.4', '3.3.9'] # Latest bugfix release in all supported minor versions
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup environment
+        run: echo "JAVA_HOME=${JAVA_HOME_8_X64}" | tee -a $GITHUB_ENV
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/
+            !~/.m2/repository/**/*-SNAPSHOT/
+            target/it-repo/
+            !target/it-repo/**/*-SNAPSHOT/
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}-mvn-${{ matrix.mvn }}
+          restore-keys: |
+            ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+      - name: Set up maven wrapper for specified version
+        run: |
+          mvn wrapper:wrapper -Dmaven=${{ matrix.mvn }}
```

Comment:
I wonder if this couldn't leverage `invoker.mavenHome` to not fail the plugin's build in case of incompatibilities.

Reply:
Not sure I'm following you - you're suggesting to give `invoker.mavenHome` the newly downloaded path, so that the plugin still builds with the same maven version, but we run with a different one?

I can't promise that the approach that I added works in all cases, but this is the test that I originally added to the j2cl-maven-plugin when we discovered that we were not compatible with older maven versions, and it reproduced the issue cleanly, and seemed to confirm the fix in a way that was consistent with the user environment. It is my understanding that regardless of maven version (A) used to build the plugin, the plugin's pom can declare its own maven dependencies (B) which will be used to compile it (and generate plugin metadata?), but it is the runtime maven (C) that actually determines the version of the maven apis... as opposed to the non-maven dependencies, which will be respected from B. I don't fully understand how conflicts between B and C are resolved, but A doesn't seem to be involved for dependencies, at least as far as I can tell?

Assuming I understand you, I'll give your suggestion a try and see if it makes a difference here at least.